spring-xd-dirt/src/main/java/org/springframework/xd/dirt/plugins/spark/MessageBusReceiver.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.xd.dirt.plugins.spark;
+
+import java.io.Serializable;
+import java.util.Properties;
+
+import org.apache.spark.storage.StorageLevel;
+import org.apache.spark.streaming.receiver.Receiver;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.springframework.context.ConfigurableApplicationContext;
+import org.springframework.integration.channel.DirectChannel;
+import org.springframework.messaging.Message;
+import org.springframework.xd.dirt.integration.bus.MessageBus;
+
+/**
+ * Spark {@link Receiver} implementation that binds to the MessageBus as a consumer.
+ * 
+ * @author Mark Fisher
+ * @author Ilayaperumal Gopinathan
+ */
+class MessageBusReceiver extends Receiver {
```

Comment:
Package name is a sufficient qualifier here, although we might want to add a "streaming" subpackage. For the record, those classes that *do* contain Spark in their name are for the extension points of XD infrastructure itself, typically subclasses or interface impls where the name itself should include qualification (e.g. the Plugin impl and the Module impl), whereas the classes that are merely supporting those should have simple names - and live in appropriate packages, especially these non-public classes (internal support classes).